File: CatsDemo_OutsideInTDD_DotNetCore/tests/Interactors.Tests/CreateCat/CreateCatInteractorTests.cs
using Interactors.CreateCat;
using System;
using Xunit;

namespace Interactors.Tests
{
    public class CreateCatInteractorTests
    {
        [Fact]
        public void ExecuteShouldCallThePresenterPassingTheRequiredParameters()
        {
            // arrange
            var presenter = new FakeCreateCatPresenter();
            var request = new CreateCatRequest("Kang Kang", "The Great");
            var interactor = new CreateCatInteractor(presenter);

            // act
            interactor.Execute(request);

            // assert
            Assert.Equal("Kang Kang", presenter.TheRequestPassedAsParameterToTheExecuteMethod.Name);
            Assert.Equal("The Great", presenter.TheRequestPassedAsParameterToTheExecuteMethod.Title);
        }
    }
}

File: practice-2017-09-30-android/MyApplication/domain/src/test/java/com/example/jboy/interactors/createcat/CreateCatInteractorTests.java
package com.example.jboy.interactors.createcat;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by Dev on 30/09/2017.
 */

public class CreateCatInteractorTests {

    FakeCatsRepository repository = new FakeCatsRepository();

    @Test
    public void shouldReturnCorrectResponse() {
        // arrange
        CreateCatInteractor interactor = new CreateCatInteractor(repository);
        CreateCatRequest request = new CreateCatRequest();
        request.name = "Kang Kang";
        request.title = "The Great";

        // act
        CreateCatResponse response = interactor.execute(request);

        // assert
        assertNotNull(response);
        assertNotNull(response.id);
        assertEquals(request.name, response.name);
        assertEquals(request.title, response.title);
    }

    @Test
    public void shouldSaveDataToTheDatabase() {
        // arrange
        CreateCatInteractor interactor = new CreateCatInteractor(repository);
        CreateCatRequest request = new CreateCatRequest();
        request.name = "Kang Kang";
        request.title = "The Great";

        // act
        CreateCatResponse response = interactor.execute(request);

        // assert
        assertTrue(repository.theSaveMethodWasCalled());
        assertNotNull(repository.theCatEntityThatWasReceivedByTheSaveMethod());
        assertNotNull(repository.theCatEntityThatWasReceivedByTheSaveMethod().getId());
        assertEquals(request.name, repository.theCatEntityThatWasReceivedByTheSaveMethod().getName());
        assertEquals(request.title, repository.theCatEntityThatWasReceivedByTheSaveMethod().getTitle());
    }
}
